### PR TITLE
[Merged by Bors] - Document need for port 9000 to be open (fix #730)

### LINKED ISSUE #730: Document the need to open port 9000
I had to open port 9000 (for both TCP and UDP traffic) on my router for synching to start.

### PR DESCRIPTION
**Pull request by JustinDrake**

### CODE REVIEW

**Comment by paulhauner** [comment]
I'm happy to merge this @JustinDrake, I notice it's in draft though. Are you happy for me to merge?

**Comment by JustinDrake** [comment]
> I'm happy to merge this @JustinDrake

Wait, @AgeManning thinks opening port 9000 [is not necessary](https://github.com/sigp/lighthouse/issues/730) 😂"Regardless of whether you are behind a NAT or not, you should be able to connect to any peer who has their ports open."

Who is right?

**Comment by AgeManning** [comment]
You do not need to open these ports. I'm very confident this was poor timing when you were connecting your node.

Although I think we should recommend it in the docs, as the more peers that port forward the default ports, the more peers people will be able to connect to.

**Comment by zedt3ster** [comment]
I've had the same behaviour as described by @JustinDrake in #730 on the same day and it definitely wasn't related to NATing for me (I could sync just fine the day before and the day after without having to open ports on my router).

**Comment by AgeManning** [comment]
I think the bootnodes were down so Justin could not sync over the bootnodes. However his IP must have been registered in the DHT somehow. Once port forwarding his ports, other peers on the network would have discovered and connected to his node and he would have begun syncing. 

The default behaviour (without port forwarding) is to connect to the bootnodes, discover other nodes and connect to all other nodes who are available on the network and participate in sync/rpc. Ports do not need to be forwarded for this behaviour.

**Comment by paulhauner** [comment]
@AgeManning has indicated to me that he's confident that you don't need the ports open to sync.

Closing this for now. Feel free to open if you disagree.

**Comment by MysticRyuujin** [comment]
I'm confused how other nodes are suppose to "connecting to all other nodes who are available" if nobody opens port 9000? Not documenting it at all (i.e. suggesting users do it) seems counter intuitive to the health of the network and the ability for random beacon nodes to peer with each other who are sitting behind stateful firewalls (basically every home user).

Prysm has a whole section for [Improve peer-to-peer connectivity](https://docs.prylabs.network/docs/prysm-usage/p2p-host-ip) where they explicitly call out the Port Forwarding options.

I feel like not documenting it at all is a disservice. Your own documentation says:
> In order to be a good peer and serve other peers you should expose port 9000. Use the -p flag to do this:
https://lighthouse-book.sigmaprime.io/docker.html?highlight=9000#ports

I think it makes sense to at least have a small section similar to Prysm that says "Hey, you don't need to do this, but it does help the network."

The whole reason I'm here making this comment is I was trying to figure out if I should add udp to the port forward rule on my docker.

**Comment by AgeManning** [comment]
> Although I think we should recommend it in the docs, as the more peers that port forward the default ports, the greater the connectivity amongst peers.

I agree we should document it more. I've opened this issue to add it to the docs. 

> "Regardless of whether you are behind a NAT or not, you should be able to connect to any peer who has their ports open

> "connect to all other nodes who are available"

As I said above, you can connect to all peers who have their ports open. So in this context "who are available" are peers that have port forwards. If nobody opens port 9000, no one is available. 

We've added UPnP support, so those with routers that support UPnP and it is enabled, should automatically have their ports forward.

**Comment by AgeManning** [comment]
We should add a section to our docs about port forwarding and UPnP support

**Review by AgeManning — APPROVED** [review]

File: book/src/advanced_networking.md [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+stores. You node will also have to handle and manage the gossip and extra
+bandwidth that comes from having these extra peers. Having a non-responsive
+node (due to overloading of connected peers), degrades the network as a whole.
+
+It is often the belief that a higher peer counts will improve sync times.
+Beyond a handful of peers, this is not true. On all current tested networks,
+the bottleneck for syncing is not the network bandwidth of downloading blocks,
+rather it is the CPU load of processing the blocks themselves. Most of the
+time, the network is idle, waiting for blocks to be processed. Having a very
+large peer count, will not speed up sync.
```
Comment (paulhauner) [inline book/src/advanced_networking.md:36]:
```suggestion
large peer count will not speed up sync.
```

File: book/src/advanced_networking.md [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+For these reasons, we recommend user's do not modify the `--target-peer` count
```
Comment (paulhauner) [inline book/src/advanced_networking.md:38]:
```suggestion
For these reasons, we recommend users do not modify the `--target-peer` count
```

**Review by paulhauner — APPROVED** [review]
Approved modulo these few typos.

Nice write up.

**Pull request closed without merging**